feat: implement view transition functionality (#2175)
jackyzha0/quartz: docs/features/View Transitions.md

## Customization

Quartz assigns appropriate `view-transition-name` properties to its component, allowing you to create custom animations for different parts of the page.

### Styling Transitions

You can customize the behavior of these transitions by adding CSS to your `quartz/styles/custom.scss` file. For example:

```scss title="quartz/styles/custom.scss"
::view-transition-old(root),
::view-transition-new(root) {
  animation-duration: 0.5s;
}
```

For more advanced customization options, see the [MDN documentation on customizing view transition animations](https://developer.mozilla.org/en-US/docs/Web/API/View_Transition_API/Using#customizing_your_animations).

### Programmatic Transitions

When [[creating components|creating your own components]], you can use the `startViewTransition` utility function to wrap DOM updates in a view transition. This ensures your custom scripts respect your view transition settings.

```typescript
import { startViewTransition } from "./scripts/util"

// Wrap your DOM update in a view transition
startViewTransition(() => {
  // Your DOM updates here
  element.textContent = "New content"
})
```

> The `startViewTransition` function will automatically use the View Transitions API if it's supported and enabled, or fall back to immediate execution otherwise.

Reviewer: ```suggestion
The `startViewTransition` function will automatically use the View Transitions API if the browser is supported, otherwise it will fallback to eager execution.
```
